internal/digitallpa/digital_lpa.go

```diff
@@ -0,0 +1,172 @@
+package digitallpa
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"github.com/ministryofjustice/opg-search-service/internal/response"
+)
+
+const AliasName = "digital_lpa"
+
+type Person struct {
+	Firstnames string  `json:"firstNames"`
+	Surname    string  `json:"surname"`
+	Address    Address `json:"address"`
+}
+
+type Donor struct {
+	Person
+	Dob string `json:"dob"`
+}
+
+type Attorney struct {
+	Person
+	Dob string `json:"dob"`
+}
+
+type Address struct {
+	Line1    string `json:"line1"`
+	Line2    string `json:"line2"`
+	Line3    string `json:"line3"`
+	Postcode string `json:"postcode"`
+}
+
+type DigitalLpa struct {
+	Uid                 string     `json:"uId"`
+	LpaType             string     `json:"lpaType"`
+	Donor               Donor      `json:"donor"`
+	CertificateProvider Person     `json:"certificateProvider"`
+	Attorneys           []Attorney `json:"attorneys"`
+}
+
+func (d DigitalLpa) Id() string {
+	return d.Uid
+}
+
+func (d DigitalLpa) Validate() []response.Error {
+	var errs []response.Error
+
+	if len(d.Uid) == 0 {
+		errs = append(errs, response.Error{
+			Name:        "uId",
+			Description: "field is empty",
+		})
+	}
+
+	return errs
+}
+
+func IndexConfig() (name string, config []byte, err error) {
+	textField := map[string]interface{}{"type": "text"}
+	searchableTextField := map[string]interface{}{"type": "text", "copy_to": "searchable"}
+	keywordField := map[string]interface{}{"type": "keyword"}
+
+	digitalLpaConfig := map[string]interface{}{
+		"settings": map[string]interface{}{
+			"number_of_shards":   1,
+			"number_of_replicas": 1,
+			"refresh_interval":   "1s",
+			"analysis": map[string]interface{}{
+				"filter": map[string]interface{}{
+					"whitespace_remove": map[string]interface{}{
+						"type":        "pattern_replace",
+						"pattern":     " ",
+						"replacement": "",
+					},
+				},
+				"analyzer": map[string]interface{}{
+					"default": map[string]interface{}{
+						"tokenizer": "whitespace",
+						"filter":    []string{"asciifolding", "lowercase"},
+					},
+					"no_space_analyzer": map[string]interface{}{
+						"tokenizer": "keyword",
+						"filter":    []string{"whitespace_remove", "lowercase"},
+					},
+				},
+			},
+		},
+		"mappings": map[string]interface{}{
+			"properties": map[string]interface{}{
+				"searchable": textField,
+				"uid":        searchableTextField,
+				"lpaType":    textField,
+				"donor": map[string]interface{}{
+					"properties": map[string]interface{}{
+						"firstnames": searchableTextField,
```

Comment:
These need to match the casing of the JSON field names, same for the other `firstNames` below.
```suggestion
				"uId":        searchableTextField,
				"lpaType":    textField,
				"donor": map[string]interface{}{
					"properties": map[string]interface{}{
						"firstNames": searchableTextField,
```